odoo/addons/base/models/qweb.py

```diff
@@ -840,822 +629,545 @@ def _compile_static_node(self, el, options):
-                        )],
-                        value=item[1]
-                    ))
-                elif item:
-                    # t_attrs.update($item)
-                    body.append(ast.Expr(ast.Call(
-                        func=ast.Attribute(
-                            value=ast.Name(id='t_attrs', ctx=ast.Load()),
-                            attr='update',
-                            ctx=ast.Load()
-                        ),
-                        args=[item],
-                        keywords=[],
-                        starargs=None, kwargs=None
-                    )))
-
+                code.append(self._indent("""attrs = {}""", indent))
+            code.extend(self._compile_static_attributes(el, options, indent))
+            code.extend(self._compile_dynamic_attributes(el, options, indent))
         if attr_already_created:
-            # tagName = $el.tag
-            body.append(ast.Assign(
-                targets=[ast.Name(id='tagName', ctx=ast.Store())],
-                value=ast.Str(el.tag))
-            )
-            body.extend(self._append_attributes())
+            code.append(self._indent(f'''tagName = "{self._compile_str(el.tag)}"''', indent))
+            code.extend(self._compile_attributes(options, indent))
+        return code
 
-        return body
-
-    def _compile_tag(self, el, content, options, attr_already_created=False):
+    def _compile_tag_open(self, el, options, indent, attr_already_created=False):
         """ Compile the tag of the given element into a list of AST nodes. """
         extra_attrib = {}
         if not el.nsmap:
             unqualified_el_tag = el_tag = el.tag
         else:
             # Etree will remove the ns prefixes indirection by inlining the corresponding
             # nsmap definition into the tag attribute. Restore the tag and prefix here.
-            # Note: we do not support namespace dynamic attributes.
+            # Note: we do not support namespace dynamic attributes, we need a default URI
+            # on the root and use attribute directive t-att="{'xmlns:example': value}".
             unqualified_el_tag = etree.QName(el.tag).localname
             el_tag = unqualified_el_tag
             if el.prefix:
-                el_tag = '%s:%s' % (el.prefix, el_tag)
+                el_tag = f'{el.prefix}:{el_tag}'
 
             # If `el` introduced new namespaces, write them as attribute by using the
             # `extra_attrib` dict.
             for ns_prefix, ns_definition in set(el.nsmap.items()) - set(options['nsmap'].items()):
                 if ns_prefix is None:
                     extra_attrib['xmlns'] = ns_definition
                 else:
-                    extra_attrib['xmlns:%s' % ns_prefix] = ns_definition
+                    extra_attrib[f'xmlns:{ns_prefix}'] = ns_definition
+
+        code = []
+        if unqualified_el_tag != 't':
+            self._appendText("<{}{}".format(
+                self._compile_str_html(el_tag),
```

Comment:
There are cases where it's necessary to HTML-escape tagnames?

Reply:
I think about the namespacing but I'm not sure

Reply:
I don't think that's an issue, namespace prefixes should be valid names, always.